### PR TITLE
Upload native packages

### PR DESCRIPTION
**Pull request by timcharper**
Also:

- assert valid tag
- assert prior release is direct ancestor
- assert target tag has new commits since prior release

JIRA Issues: MARATHON-7829

### CODE REVIEW

File: ci/README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ The `jenkins` target is excuted on every branch build by jenkins. It runs
   * `provision.killStaleTestProcesses()`
   * `provision.installMesos()`
   * `compileAndTest()`
-  * `createPackages()`
+  * `build()`
 
 The `phabricator` target runs the `jenkins` pipeline followed by Phabricator
```
Comment (alenkacz) [inline ci/README.md]: I know it is not connected to your changes but this part of readme is probably outdated because it mentions phabricator... ?

**Comment by timcharper** [comment]
Oh, yes :)

File: ci/pipeline [hunk 2]
```diff
@@ -273,9 +285,79 @@ def release(requestVersion: String, gitSha: String, latest: Boolean = false): Un
...
+/**
+  * Given tagName v1.5.1, and rev DEADBEEF, we would assert that v1.5.0 is a direct ancestor of rev DEADBEEF
+  *
+  * Also, we assert that rev contains new commits since the last release. If the proposal is to tag v1.5.1 with same
+  * revision as v1.5.0, then we fail, saying that there v1.5.0 points to the commit porposed for v1.5.1
+  *
+  * @param tagName The tag name / version we are going to tag. eg v1.5.1
+  * @param gitSha The commit sha / rev that will be tagged.
+  */
+def assertPriorVersionIsAncestor(tagName: String, gitSha: String): Unit = {
```
Comment (jeschkies) [inline ci/pipeline]: Why is this required?

File: ci/pipeline [hunk 3]
```diff
@@ -273,9 +285,79 @@ def release(requestVersion: String, gitSha: String, latest: Boolean = false): Un
...
+  })
+  val pkgserverHost = sys.env.getOrElse("PKG_SSH_HOST", {
+    throw new IllegalStateException("PKG_SSH_HOST environment variable must be set")
+  })
+
+  // Note - the key is expected to be provided via an SSH agent
+  utils.printStageTitle(s"Uploading native packages")
+  %("rsync", "-avz",
+    (pwd / 'target / 'packages) + "/",
+    s"${pkgserverUser}@${pkgserverHost}:repo/incoming/marathon-${tagName}/")
```
Comment (jeschkies) [inline ci/pipeline]: So we first rsync the packages and then move them on the server? Where is the unstable channel defined?

File: ci/pipeline [hunk 4]
```diff
@@ -273,9 +285,79 @@ def release(requestVersion: String, gitSha: String, latest: Boolean = false): Un
...
+}
+
 def withoutVersionPrefix(releaseVersion: String): String = {
     if(releaseVersion.toLowerCase.startsWith("v"))
       releaseVersion.substring(1)
     else
       releaseVersion
 }
+
+def uploadLinuxPackagesToRepos(tagName: String): Unit = {
```
Comment (jeschkies) [inline ci/pipeline]: This does not seem to be called anywhere. Should we upload each master build? As far as I understood we agreed to have the master builds on the unstable release channel.

File: ci/pipeline (hunk 2 above)
Comment (kensipe) [reply ci/pipeline]: this ensures that the sha that is being released is on the correct release branch.   It is designed to prevent human error... because I understand that humans like errors so much they create them or something

File: ci/pipeline (hunk 4 above)
Comment (kensipe) [reply ci/pipeline]: that may be true... I understand that if we do release from master that should be on the unstable release channel.  I was sure or clear if we agreed we should do that.   At this point I would really like to publish our releases.

File: ci/pipeline (hunk 3 above)
Comment (kensipe) [reply ci/pipeline]: nothing yet created for the unstable channel.     Perhaps we will have agreement out of our release meeting.

File: ci/pipeline (hunk 2 above)
Comment (jeschkies) [reply ci/pipeline]: Ah, so I cannot release a 1.6 commit as 1.5? Could we express this in the comments?

File: ci/pipeline (hunk 4 above)
Comment (jeschkies) [reply ci/pipeline]: Ok, can we follow up then? Also, shouldn't this method be called in the release target?

File: ci/pipeline (hunk 2 above)
Comment (kensipe) [reply ci/pipeline]: it seems like it is there but I made a clarification statement.

File: ci/pipeline (hunk 4 above)
Comment (kensipe) [reply ci/pipeline]: it is called in the release method... line 269
